components/support/nimbus-fml/src/parser.rs

```diff
@@ -90,6 +214,184 @@ fn parse_typeref_string(input: String) -> Result<(String, Option<String>), FMLEr
     }
 }
 
+/// Collects the channel defaults of the feature manifest
+/// and merges them by channel
+///
+/// **NOTE**: defaults with no channel apply to **all** channels
+///
+/// # Arguments
+/// - `defaults`: a [`serde_json::Value`] representing the array of defaults
+///
+/// # Returns
+/// Returns a [`std::collections::HashMap<String, serde_json::Value>`] representing
+/// the merged defaults. The key is the name of the channel and the value is the
+/// merged json.
+///
+/// # Errors
+/// Will return errors in the following cases (not exhaustive):
+/// - The `defaults` argument is not an array
+fn collect_channel_defaults(
+    defaults: serde_json::Value,
+) -> Result<HashMap<String, serde_json::Value>, FMLError> {
+    let mut channel_map = HashMap::new();
+    let defaults = serde_json::from_value::<Vec<FeatureDefaults>>(defaults)?;
+    for default in defaults {
+        if let Some(channel) = default.channel {
+            if channel_map.contains_key(&channel) && default.targeting.is_none() {
+                let old_default = channel_map.get(&channel).unwrap();
+                let merged = merge_two_defaults(old_default, &default.value);
+                channel_map.insert(channel, merged);
+            } else {
+                channel_map.insert(channel, default.value);
+            }
+        // This is a default with no channel, so it applies to all channels
+        } else {
+            channel_map = channel_map
+                .into_iter()
+                .map(|(channel, old_default)| {
+                    (channel, merge_two_defaults(&old_default, &default.value))
+                })
+                .collect();
+        }
+    }
+    Ok(channel_map)
+}
+
+/// Transforms a feature definition with unmerged defaults into a feature
```

Comment:
Not sure how we feel about doc comments in this project.. happy to get rid of it if we feel it clutters the code and doesn't add much value! (wanted to be explicit that I am totally happy to throw the comments away and I'm a bit conflicted about the "when" of doc comments and how we want to use them)